packages/dds/merge-tree/src/properties.ts

```diff
@@ -126,6 +136,9 @@ export function extend<T>(
 			}
 		}
 	}
+	if (Object.entries(base).length === 0) {
+		return undefined;
```

Comment:
Does this have any effect? In our codebase, the callers don't use the return value, just base itself.